frontend/app/services/wsaddress-service.server.ts

```diff
@@ -6,191 +6,65 @@ import { getLogger } from '~/utils/logging.server';
-        message: z.string(),
-      }),
-    )
-    .optional(),
+const parseAddressResponseSchema = z.object({
+  'wsaddr:ParsedResults': z.object({
+    'nc:AddressFullText': z.string(),
+    'nc:AddressCityName': z.string(),
+    'can:ProvinceCode': z.string().optional(),
+    'nc:AddressPostalCode': z.string().optional(),
+    'nc:CountryCode': z.string(),
+    'wsaddr:AddressSecondaryUnitNumber': z.string().optional(),
+  }),
 });
 
-//TODO: Update validation schema when changing to use Interop's WSAddress endpoint instead of MSW
-const validateWSAddressResponseSchema = z.object({
-  responseType: z.string().optional(),
-  statusCode: z.string().optional(),
-  functionalMessages: z
-    .array(
-      z.object({
-        action: z.string(),
-        message: z.string(),
-      }),
-    )
-    .optional(),
-  message: z.string().optional(),
-  warnings: z.string().optional(),
+const validateAddressResponseSchema = z.object({
+  'wsaddr:ValidationResults': z.object({
+    'wsaddr:Information': z.object({
+      'wsaddr:StatusCode': z.string(),
+    }),
+  }),
 });
 
 export const getWSAddressService = moize.promise(createWSAddressService, { onCacheAdd: () => log.info('Creating new WSAddress service') });
 
 async function createWSAddressService() {
   const { INTEROP_API_BASE_URI } = getEnv();
 
-  async function correctWSAddress({ address, city, province, postalCode, country }: { address: string; city: string; province: string; postalCode: string; country: string }) {
-    const request: CorrectWSAddressRequestDTO = {
+  async function correctAddress({ address, city, province, postalCode, country }: { address: string; city: string; province: string; postalCode: string; country: string }) {
+    const searchParams = new URLSearchParams({
       addressLine: address,
       city,
       province,
       postalCode,
       country,
-      formatResult: true,
+      formatResult: 'true',
       language: 'English', // TODO confirm that we should always have this as "English"
-    };
-    const url = `${INTEROP_API_BASE_URI}/address/correction`;
-    const response = await fetch(url, {
-      method: 'POST',
-      body: JSON.stringify(request),
     });
+    const url = `${INTEROP_API_BASE_URI}/CAN/correct?${searchParams}`;
+    const response = await fetch(url);
 
     if (response.ok) {
-      return correctWSAddressResponseSchema.parse(await response.json());
+      const correctAddressResponse = correctAddressResponseSchema.parse(await response.json());
+      const correctionResults = correctAddressResponse['wsaddr:CorrectionResults'];
+      return {
+        status: correctionResults['wsaddr:StatusCode'],
+        address: correctionResults['nc:AddressFullText'],
+        city: correctionResults['nc:AddressCityName'],
+        province: correctionResults['nc:ProvinceCode'],
+        postalCode: correctionResults['nc:AddressPostalCode'],
+        country: correctionResults['nc:CountryCode'],
+      };
     }
 
     log.error('%j', {
```

Comment:
I mentioned this in other PRs, and will mention it again.. I'm not **necessarily** suggesting a change, just want to start a discussion.

I feel like a better order to these services is:

1. fetch
1. handle errors
1. post-process and return

The reason being that sometimes we want to format, tweak, or do some other post-processing with the returned data. We're not doing that now, but we might in the future, and then we're going to end up with a huge `if { /* post-processing here */ }` block right in the middle of the function.

After writing a bunch of these over the years, I find that this is better:

``` typescript
function fetchSomething() {
  const resp = fetch(..);

  if (!resp.ok) {
    // throw or something
  }

  const data = resp.json();

  //
  // validate, map, combine, tweak... or do whatever with the data here
  //

  return retVal;
}
```

Reply:
Good point and agreed.